src/tests/ftest/pool/management_race.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python3
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+import time
+import random
+from apricot import TestWithServers
+from thread_manager import ThreadManager
+from command_utils_base import CommandFailure
+
+class PoolManagementRace(TestWithServers):
+    """
+    Epic: Create system level tests that cover pool management race boundary tests and
+          functionality.
+    Testcase:
+          DAOS-8463: Pool management integrity with race boundary conditions
+    Test Class Description:
+          Start DAOS servers, create pools and delete the pool while query and list.
+    :avocado: recursive
+    """
+
+    def del_recreate_query_and_list_pools(
```

Comment:
I think it's better to separate this method to `del_recreate_pools()` and `query_list_pools()`. In general, we want each method to do just one thing.

Reply:
Merge the 2 methods is to allow the ThreadManager to run the method simultaneously and trigger race condition.